### PR TITLE
Recipients filter

### PR DESCRIPTION
**Pull request by michielderoos**
**Describe the Pull Request**

A clear and concise description of the features implemented in the PR of bugs fixed.

**Todo**

- [x] Link relevant [trello card](https://trello.com/b/PA2LhlOh/sempo-dev) or [related issue](https://github.com/teamsempo/SempoBlockchain/issues) to the pull request
- [x] Request review from relevant @person or team
- [x] QA your pull request. This includes running the app, login, testing changes etc.
- [x] Bump [backend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/config.py#L7) and/or [frontend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/app/package.json#L3) versions following Semver

### CODE REVIEW

**Comment by michielderoos** [comment]
[Add filter by for recipients](https://trello.com/c/BhRwhAmC/663-add-filter-by-for-recipients)

File: app/server/utils/transfer_filter.py [hunk 1]
```diff
@@ -74,35 +101,86 @@ def ALL_FILTERS(self):
...
+        return {
+            **get_custom_attribute_filters(distinct_sender_and_recipient=True),
+            **get_custom_attribute_filters(distinct_sender_and_recipient=False),
+            **self.transfer_filters,
+            **self.user_filters,
+            **self.tx_rx_user_filters,
+        }
+
+    @property
+    def TRANSFER_FILTERS(self): return { **self.tx_rx_user_filters, **self.transfer_filters, **get_custom_attribute_filters(distinct_sender_and_recipient=True) }
```
Comment (enjeyw) [inline app/server/utils/transfer_filter.py:179]: NIT: can we put this on two lines?

**Review by enjeyw — APPROVED** [review]
Tested and looks good to me!

Imma gonna avoid putting this one through to prod until we have hidable fields, as it makes the options pretty overwhelming - perhaps we could merge this and the upcoming PR into side-branches until they're both ready (or something equivalent, like stacked PRs)? No worries if too much effort from a merge-hell PoV. 
 I was also thinking that somehting like this:
https://codepen.io/enjeyw/pen/pobMNrR?editors=0010
Could help keep it managable (probably as a later PR, if we think is required)

**Comment by tristanhcole** [comment]
> Tested and looks good to me!
> 
> Imma gonna avoid putting this one through to prod until we have hidable fields, as it makes the options pretty overwhelming - perhaps we could merge this and the upcoming PR into side-branches until they're both ready (or something equivalent, like stacked PRs)? No worries if too much effort from a merge-hell PoV.
> I was also thinking that somehting like this:
> https://codepen.io/enjeyw/pen/pobMNrR?editors=0010
> Could help keep it managable (probably as a later PR, if we think is required)

I think the option group looks _wayyyy_ better and more usable. +1000

**Pull request merged**